controllers/serviceinstance_controller.go

```diff
@@ -120,13 +124,69 @@ func (r *ServiceInstanceReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		return r.createInstance(ctx, smClient, serviceInstance)
 	}
 
+	// Handle instance share change if needed
+	if instanceShareChangeNeedsToBeHandled(serviceInstance) {
+		if err := r.handleInstanceSharingChange(ctx, serviceInstance, smClient); err != nil {
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{Requeue: true, RequeueAfter: r.Config.PollInterval}, nil
+	}
+
 	// Update
 	if serviceInstance.Status.Ready == metav1.ConditionTrue {
 		return r.updateInstance(ctx, smClient, serviceInstance)
 	}
+
 	return ctrl.Result{}, nil
 }
 
+func instanceShareChangeNeedsToBeHandled(serviceInstance *servicesv1.ServiceInstance) bool {
```

Comment:
i think the name is misleading because you check the following:
1. is in desired state
2. is instance in ready state
3. did the sharing failed

personally when I find it too hard to name a function, I delete it :).
In this case, I suggest to delete the function and move the logic to handleInstanceSharingChange

Reply:
Lets talk about it :) i think that the function name is informative and do the job
To start handleInstanceSharingChange anyway will make the controller logic in-correct since i'm requeing there